src/cargo/util/toml.rs

```diff
@@ -109,9 +109,7 @@ pub fn to_manifest(contents: &str,
     let manifest: TomlManifest = serde_ignored::deserialize(root, |path| {
         let mut key = String::new();
         stringify(&mut key, &path);
-        if !key.starts_with("package.metadata") {
```

Comment:
It is indeed intentional! Note though that these keys are now all considered used because they're decoded directly into the manifest, so they won't show up here any more.